### ISSUE TITLE
API:n esittely

### ISSUE THREAD

**Issue by kyyberi** [Digipalvelutehdas/API-hallinta#4]
_From @kyyberi on July 19, 2015 14:16_

Kehittäjänä haluan nähdä API:sta yhdellä näkymällä kaiken oleellisen:

* Kuvaus mistä kyse, mitä tarvetta täyttää ja kenelle suunnattu (pari kappaletta)
* Tila: idea, suunniteltu, kehityksessä, beta, tuotannossa, poistumassa (+ päivämäärä + varoitus!), poistettu
* deployment historia
* perusmetriikka ja vasteajat
* Koska viimeksi muutettu
* paljonko rajapintaan kohdistuu kyselyjä
* Onko rajoitteita kyselyille esim 1000 / päivä
* Kuka on tuoteomistaja ja hänen yhteystiedot
* sandbox jos sellainen on 
* datatyypit (JSON, XML)
* autentikointi tavat 
* linkki dokumentaatioon
* linkki koodiin (github)
* linkki kuvaukseen (esim RAML)
* Linkit koodiesimerkkeihin 
* Linkit SDK paketteihin eri kielillä 
* Kommentti kenttä palautetta varten
* Lisätä omaan API listaukseen (favorites, useita listoja)

_Copied from original issue: Digipalvelutehdas/Portaali#15_

**Comment by kyyberi** [comment]
* koodin lisenssi
* käyttöehdot

**Comment by jykae** [comment]
Could API category fit into this? Public, private...

**Comment by kyyberi** [comment]
Yes, but perhaps the attribute is more like API publicity: Private - Public

**Comment by kyyberi** [comment]
Profiilin synty- ja käyttöprosessin luuranko
![palveluprofiili-apika](https://cloud.githubusercontent.com/assets/1927158/9151950/f3adf2dc-3e1e-11e5-8be6-802062abbc6f.png)
Sen sijaan että API keyt sidotaan kehittäjään, tehdään palveluprofiili johon ne liitetään. Näin palvelun siirto omistajalta toiselle on mahdollista tehdä yhtenä pakettina. Samoin profiilia voi käyttää kloonaamisessa, toiset voi tehdä oman forkin jostain palveluprofiilista. Tämä saattaisi helpottaa palvelukehitystä joissain tapauksissa.
